src/ruby/ext/grpc/rb_call.c

```diff
@@ -179,6 +179,36 @@ static VALUE grpc_rb_call_cancel(VALUE self) {
   return Qnil;
 }
 
+/* Not a part of the surface API.
+ * This is meant for internal usage by the "write thread" of grpc-ruby
+ * client-side bidi calls. It provides a way for the background write-thread
+ * to propogate failures to the main read-thread and give the user an error
+ * message. */
+static VALUE grpc_rb_call_cancel_with_bidi_write_thread_error_message(
```

Comment:
per offline conversation. 

the `const char *` type for the `description` API seems to force some memory management if surfacing the `grpc_call_cancel_with_status` string directly.

But looking into `grpc_call_cancel_with_status`, it looks like the error string that it stores is copied out from the `description` parameter anyways. (https://github.com/grpc/grpc/blob/4ea19235d72404c9fb56fcf155f7fbf8ea8ec37e/src/core/lib/surface/call.c#L648). I'm not sure if this is subject to change, but if not, perhaps the `const char * description` in `grpc_call_cancel_with_status` is overly restrictive.

cc @nicolasnoble @ncteisen